From 8a2a2fe61259dbd2c471bfe2b21670c0116e46d3 Mon Sep 17 00:00:00 2001
From: chalesyu <574249312@qq.com>
Date: Fri, 31 Dec 2021 00:00:00 +0800
Subject: [PATCH] Fix PMIC support issue on some TQC-A01 Board

TQC-A01 have a AXP805 PMIC, but on some board with
hardware issue. Sometimes, it can't be found on I2C
or RSB. So when ATF(arm trust firmware) set regulator
 from DT(device tree), It will cause hang when fail.
---
 arch/arm/dts/sun50i-h6-tanix-tx6.dts | 2 +-
 configs/tqc_a01_defconfig            | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/sun50i-h6-tanix-tx6.dts b/arch/arm/dts/sun50i-h6-tanix-tx6.dts
index ccf777d0fd..1b295f485e 100644
--- a/arch/arm/dts/sun50i-h6-tanix-tx6.dts
+++ b/arch/arm/dts/sun50i-h6-tanix-tx6.dts
@@ -118,7 +118,7 @@
 	vmmc-supply = <&reg_vcc3v3>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;
 	bus-width = <4>;
-	status = "disabled";
+	status = "okay";
 };
 
 &mmc2 {
diff --git a/configs/tqc_a01_defconfig b/configs/tqc_a01_defconfig
index 9e99f60c3d..ddc4d3431f 100644
--- a/configs/tqc_a01_defconfig
+++ b/configs/tqc_a01_defconfig
@@ -1,6 +1,7 @@
 CONFIG_ARM=y
 CONFIG_ARCH_SUNXI=y
-CONFIG_DEFAULT_DEVICE_TREE="sun50i-h6-tqc-a01"
+#CONFIG_DEFAULT_DEVICE_TREE="sun50i-h6-tqc-a01"
+CONFIG_DEFAULT_DEVICE_TREE="sun50i-h6-tanix-tx6"
 CONFIG_SPL=y
 CONFIG_MACH_SUN50I_H6=y
 CONFIG_SUNXI_DRAM_H6_LPDDR3=y
-- 

